class2.md

```diff
@@ -0,0 +1,287 @@
+* Out of those 165 cases, the classifier predicted "yes" 110 times, and "no" 55 times.
+* In reality, 105 patients in the sample have the disease, and 60 patients do not.
+
+Confusion matrix terminology:
+* true positives (TP): These are cases in which we predicted yes (they have the disease), and they do have the disease.
+* true negatives (TN): We predicted no, and they don't have the disease.
+* false positives (FP): We predicted yes, but they don't actually have the disease. (Also known as a "Type I error.")
+* false negatives (FN): We predicted no, but they actually do have the disease. (Also known as a "Type II error.")
+
+There are many different metrics that can be derived from a confusion matrix. Two of the most basic metrics are *accuracy* and the *error rate* of a classifier.
+* Accuracy: A measure of how often the machine guesses correctly
+  * `(TP+TN)/total = (100+50)/165 = 0.91`
+* Error rate: A measure of how often the machine guesses incorrectly
+  * `(FP+FN)/total = (10+5)/165 = 0.09`
+  * Equivalent to `1 - Accuracy`
+
+You could further interrogate the above problem by evaluating the sensitivity of the algorithm. Sensitivity measures the true positive rate and would show that the algorithm has 100% sensitivity for cat pictures and 0% sensitivity for dog pictures.
+
+### Regression
+
+<p align="center">
+  <img width="400" alt="" src="images/regression.jpg">
+</p>
+
+Regression is a form of supervised learning that predicts a continuous numerical value. 
+* Linear regression: Aims to draw a straight best fit line through a field of data points. 
+* Polynomial regression: Aims to draw a curved best fit line through a field of data points.
+
+### Linear regression (AKA ordinary least squares)
+
+<p align="center">
+  <img width="400" alt="" src="images/meansqerror.png">
+</p>
+
+This is the most simple form of regression. The linear regression aims to fit a straight line onto the data that minimizes the mean squared error between the predicted value and the true output value. The mean squared error is the sum of the squared differences between the predictions and the true values, divided by the number of samples.
+
+You can also use polynomials to model curvature and interaction effects.
+
+Ordinary least squares regressions are very sensitive to outliars!
+
+### Testing a regression algorithm
+
+Similarly to classification algorithms, there are many different metrics that can be used when testing and tuning a regression algorithm.
+
+Some common choices include:
+* Mean Squared Error (MSE)
+* Root Mean Squared Error (RMSE)
+* Mean Absolute Error (MAE)
+* R Squared (R²)
+* Adjusted R Squared (R²)
+* Mean Square Percentage Error (MSPE)
+* Mean Absolute Percentage Error (MAPE)
+* Root Mean Squared Logarithmic Error (RMSLE)
+
+Which metric you will use will be highly dependent on the context in which you're evaluating your algorithm.
+
+#### R Squared (R^2) is one of the simplest evaluations of a regression
```

Comment:
Same issue here as above: I don't think these need to be separate subheadings, and should include more narrative to describe how they relate